packages/app/pages/queue.tsx

```diff
@@ -0,0 +1,196 @@
+import styled from "styled-components";
+import {
+  Role,
+  Question,
+  OpenQuestionStatus,
+  ClosedQuestionStatus,
+  QuestionType,
+  QuestionStatus,
+  UserCourse,
+} from "@template/common";
+import QuestionForm from "../components/Queue/QuestionForm";
+import QueueList from "../components/Queue/QueueList";
+import StudentPopupCard from "../components/Queue/StudentPopupCard";
+import {
+  useCallback,
+  useState,
+  useContext,
+  useEffect,
+  useMemo,
+  Fragment,
+} from "react";
+import useSWR from "swr";
+import { API } from "@template/api-client";
+import { ProfileContext } from "../contexts/ProfileContextProvider";
+
+// TODO: replace this with profile role from endpoint
+const ROLE: Role = Role.TA;
+
+const queueId: number = 169;
+
+const Container = styled.div`
+  margin: 32px 64px;
+  @media (max-width: 768px) {
+    margin: 32px 24px;
+  }
+`;
+
+interface QueueProps {}
+
+export default function Queue({}: QueueProps) {
+  const [isJoining, setIsJoining] = useState<boolean>(false);
+  const [openPopup, setOpenPopup] = useState<boolean>(false);
+  const { profile } = useContext(ProfileContext);
+  const [course, setCourse] = useState<UserCourse>(null);
+  const [questions, setQuestions] = useState<Question[]>([]);
+  const [queueId, setQueueId] = useState<number>(null);
+  const [questionDraftId, setQuestionDraftId] = useState<number>(null);
+  const [currentQuestion, setCurrentQuestion] = useState<Question>(null);
```

Comment:
There's a lot going on in this component which is not necessarily a bad thing but it can make it hard to easily understand what is going on. Not an issue for this PR but in the future, if we add more complexity, it might be worth considering separating this out into a `TAQueue` component and a `StudentQueue` component which could be rendered from a conditional in the parent `Queue` component.

Reply:
I think it definitely makes sense to separate this out into two components. Perhaps, it would make a good smaller ticket for the future?